sys/net/network_layer/ng_ipv6/nc/ng_ipv6_nc.c

```diff
@@ -43,27 +44,40 @@ ng_ipv6_nc_t *_find_free_entry(void)
     return NULL;
 }
 
-int ng_ipv6_nc_add(kernel_pid_t iface, const ng_ipv6_addr_t *ipv6_addr,
-                   const void *l2_addr, size_t l2_addr_len, uint8_t flags)
+ng_ipv6_nc_t *ng_ipv6_nc_add(kernel_pid_t iface, const ng_ipv6_addr_t *ipv6_addr,
+                             const void *l2_addr, size_t l2_addr_len, uint8_t flags)
 {
     if (ipv6_addr == NULL) {
         DEBUG("ipv6_nc: address was NULL\n");
-        return -EFAULT;
+        return NULL;
     }
 
     if ((l2_addr_len > NG_IPV6_NC_L2_ADDR_MAX) || (iface == KERNEL_PID_UNDEF) ||
         ng_ipv6_addr_is_unspecified(ipv6_addr)) {
-        return -EINVAL;
+        DEBUG("ipv6_nc: invalid parameters\n");
+        return NULL;
     }
 
     for (int i = 0; i < NG_IPV6_NC_SIZE; i++) {
         if (ng_ipv6_addr_equal(&(ncache[i].ipv6_addr), ipv6_addr)) {
-            DEBUG("ipv6_nc: Address %s already registered\n",
+            DEBUG("ipv6_nc: Address %s already registered. Update",
                   ng_ipv6_addr_to_str(addr_str, ipv6_addr, sizeof(addr_str)));
-            return -EADDRINUSE;
+
+            if ((l2_addr != NULL) && (l2_addr_len > 0)) {
+                DEBUG(" to L2 address %s",
+                      ng_netif_addr_to_str(addr_str, sizeof(addr_str),
+                                           l2_addr, l2_addr_len));
+
+                memcpy(&(ncache[i].l2_addr), l2_addr, l2_addr_len);
+                ncache[i].l2_addr_len = l2_addr_len;
+                ncache[i].flags = flags;
+                DEBUG(" with flags = 0x%0x\n", flags);
+
+            }
+            return &ncache[i];
         }
 
-        if (ncache[i].iface == KERNEL_PID_UNDEF) {
+        if (ng_ipv6_addr_is_unspecified(&(ncache[i].ipv6_addr))) {
```

Comment:
Oh sorry, I missed that one! If you want to, I can also do my best with a fixup PR and let you have a look at it, to minimize your probably huge list of TODOs :)